src/compiler/checker.ts

```diff
@@ -9072,6 +9075,34 @@ namespace ts {
             return type;
         }
 
+        /**
+         * Returns if a type is or consists of a JSLiteral object type
+         */
+        function isJSLiteralType(type: Type): boolean {
+            if (noImplicitAny) {
+                return false; // Flag is meaningless under `noImplicitAny` mode
+            }
+            // In addition to objects which are directly literals,
+            // * unions where every element is a jsliteral
+            // * intersections where at least one element is a jsliteral
+            // * and instantiable types constrained to a jsliteral
+            // Should all count as literals and not print errors on access or assignment of imaginary properties.

```

Comment:
quibble: these properties are not "imaginary". They are "possibly existing" properties.

Seriously, though, I'd say "expando properties". I also don't think that referencing the old implementation helps, so I'd delete that line or at least change it to "This mirrors the behaviour of index signatures".